pkg/chunk/storage/factory.go

```diff
@@ -103,29 +113,58 @@ func NewStore(cfg Config, storeCfg chunk.StoreConfig, schemaCfg chunk.SchemaConf
 	return stores, nil
 }
 
-func nameToStorage(name string, cfg Config, schemaCfg chunk.SchemaConfig) (chunk.StorageClient, error) {
+// NewIndexClient makes a new index client of the desired type.
+func NewIndexClient(name string, cfg Config, schemaCfg chunk.SchemaConfig) (chunk.IndexClient, error) {
 	switch name {
 	case "inmemory":
-		return chunk.NewMockStorage(), nil
-	case "aws":
-		return aws.NewS3StorageClient(cfg.AWSStorageConfig, schemaCfg)
-	case "aws-dynamo":
+		store := chunk.NewMockStorage()
+		return store, nil
+	case "aws", "aws-dynamo", "dynamo":
 		if cfg.AWSStorageConfig.DynamoDB.URL == nil {
 			return nil, fmt.Errorf("Must set -dynamodb.url in aws mode")
 		}
 		path := strings.TrimPrefix(cfg.AWSStorageConfig.DynamoDB.URL.Path, "/")
 		if len(path) > 0 {
 			level.Warn(util.Logger).Log("msg", "ignoring DynamoDB URL path", "path", path)
 		}
-		return aws.NewStorageClient(cfg.AWSStorageConfig.DynamoDBConfig, schemaCfg)
+		return aws.NewDynamoDBStorageClient(cfg.AWSStorageConfig.DynamoDBConfig, schemaCfg)
 	case "gcp":
 		return gcp.NewStorageClientV1(context.Background(), cfg.GCPStorageConfig, schemaCfg)
-	case "gcp-columnkey":
+	case "gcp-columnkey", "bigtable":
 		return gcp.NewStorageClientColumnKey(context.Background(), cfg.GCPStorageConfig, schemaCfg)
 	case "cassandra":
 		return cassandra.NewStorageClient(cfg.CassandraStorageConfig, schemaCfg)
+	default:
+		return nil, fmt.Errorf("Unrecognized storage client %v, choose one of: aws, gcp, cassandra, inmemory", name)
+	}
+}
+
+// NewObjectClient makes a new ObjectClient of the desired types.
+func NewObjectClient(name string, cfg Config, schemaCfg chunk.SchemaConfig) (chunk.ObjectClient, error) {
```

Comment:
In this commit you have used "ChunkClient" 21 times and "ObjectClient" 48 times.
I can't figure out the distinction - if there isn't one then I think it would be better to settle on one term.

Reply:
Started with ChunkClient but lint complained of stutter for went with ObjectClient, but missed a few.  Have fixed that now.